### PR TITLE
feat: liquidation fund

### PR DESCRIPTION
**Pull request by nieldeckx**
Ticket: https://github.com/arkadiko-dao/arkadiko/issues/240

### CODE REVIEW

**Comment by nieldeckx** [comment]
**arkadiko-liquidation-fund-v1-1.clar**
Keeping it as simple as possible:
- Users can deposit STX
- Fund controller can transfer contract STX and DIKO to given wallet
- Need to track deposits and earnings off chain
- Encapsulate key functions (swap, LP, stake, bid, redeem lot)
- Helper functions to provide/stake LP

**Review by philipdesmedt — COMMENTED** [review]
Did a first review, have one question

File: clarity/contracts/arkadiko-liquidation-fund-v1-1.clar [hunk 1]
```diff
@@ -0,0 +1,347 @@
...
+  (begin
+    (asserts! 
+      (or
+        (is-eq tx-sender (var-get fund-controller))
+        (is-eq tx-sender (as-contract tx-sender))
+      )
+      (err ERR-NOT-AUTHORIZED)
+    )
+
+    (try! (swap-stx-to-usda (/ stx-amount u2) u0))
```
[the remark this answers is not in the document]
Comment (philipdesmedt) [reply clarity/contracts/arkadiko-liquidation-fund-v1-1.clar:106]: Yeah, you're right :D

**Comment by philipdesmedt** [comment]
I can't get the tests fixed on this branch, not sure what's wrong. Going to merge and see what happens on master.

**Pull request merged**